src/proxy.rs

```diff
@@ -15,7 +14,11 @@ use {into_url, IntoUrl, Url};
 ///
 /// ```rust
 /// # fn run() -> Result<(), Box<::std::error::Error>> {
-/// let proxy = reqwest::Proxy::http("https://secure.example")?;
+/// let mut proxy = reqwest::Proxy::http("https://secure.example")?;
+/// // proxy.set_authorization(Basic {
+/// //     username: "John Doe".into(),
+/// //     password: Some("Agent1234".into()),
+/// // });
```

Comment:
This probably shouldn't show the user some code that they can't actually use :)

Reply:
Yes I plan to have it for merge though.